src/overlays/actors/ovl_En_Go/z_en_go.h

```diff
@@ -7,20 +7,48 @@
 struct EnGo;
 
 typedef void (*EnGoActionFunc)(struct EnGo*, PlayState*);
-typedef u16 (*callback1_80A3ED24)(PlayState*, struct EnGo*);
-typedef s16 (*callback2_80A3ED24)(PlayState*, struct EnGo*);
-
-// WIP type docs
-// /* 0x00 */ GORON1_CITY_LINK,
-// /* 0x10 */ GORON1_FIRE_GENERIC,
-// /* 0x20 */ GORON1_DMT_DC_ENTRANCE,
-// /* 0x30 */ GORON1_DMT_ROLLING_SMALL,
-// /* 0x40 */ GORON1_DMT_BOMB_FLOWER,
-// /* 0x50 */ GORON1_CITY_ENTRANCE,
-// /* 0x60 */ GORON1_CITY_ISLAND,
-// /* 0x70 */ GORON1_CITY_LOST_WOODS,
-// /* 0x80 */ // Not Used
-// /* 0x90 */ GORON1_DMT_BIGGORON,
+
+typedef enum GoronLimb {
+    /*  0 */ GORON_LIMB_NONE, // skeleton itself
+    /*  1 */ GORON_LIMB_ROOT,
+    /*  2 */ GORON_LIMB_WAIST, // drives bottom submesh
+    /*  3 */ GORON_LIMB_LEGS,
+    /*  4 */ GORON_LIMB_LEFT_THIGH,
+    /*  5 */ GORON_LIMB_LEFT_SHIN,
+    /*  6 */ GORON_LIMB_LEFT_FOOT,
+    /*  7 */ GORON_LIMB_RIGHT_THIGH,
+    /*  8 */ GORON_LIMB_RIGHT_SHIN,
+    /*  9 */ GORON_LIMB_RIGHT_FOOT,
+    /* 10 */ GORON_LIMB_TORSO, // drives top submesh
+    /* 11 */ GORON_LIMB_LEFT_ARM,
+    /* 12 */ GORON_LIMB_LEFT_FOREARM,
+    /* 13 */ GORON_LIMB_LEFT_HAND,
+    /* 14 */ GORON_LIMB_RIGHT_ARM,
+    /* 15 */ GORON_LIMB_RIGHT_FOREARM,
+    /* 16 */ GORON_LIMB_RIGHT_HAND,
+    /* 17 */ GORON_LIMB_HEAD,
+    /* 18 */ GORON_LIMB_MAX
+} GoronLimb;
+
+typedef enum EnGoType {
+    ENGO_TYPE_CITY_LINK         = (0 << 4),
+    ENGO_TYPE_FIRE_GENERIC      = (1 << 4),
+    ENGO_TYPE_DMT_DC_ENTRANCE   = (2 << 4),
+    ENGO_TYPE_DMT_ROLLING_SMALL = (3 << 4),
+    ENGO_TYPE_DMT_BOMB_FLOWER   = (4 << 4),
+    ENGO_TYPE_CITY_ENTRANCE     = (5 << 4),
+    ENGO_TYPE_CITY_ISLAND       = (6 << 4),
+    ENGO_TYPE_CITY_LOST_WOODS   = (7 << 4),
+    ENGO_TYPE_DMT_BIGGORON      = (9 << 4)
+} EnGoType;
+
+#define ENGO_GET_PATH_INDEX(this) PARAMS_GET_U((this)->actor.params, 0, 4)
+#define ENGO_GET_TYPE(this) PARAMS_GET_NOSHIFT((this)->actor.params, 4, 4)
+#define ENGO_CAGED_SWITCH_FLAG(this) PARAMS_GET_NOMASK((this)->actor.params, 8)
+
+#define ENGO_PATH_NONE NBITS_TO_MASK(4)
+#define ENGO_IS_CAGE_OPEN(this, play) Flags_GetSwitch(play, ENGO_CAGED_SWITCH_FLAG(this))
```

Comment:
```suggestion
```

I don't think it serves much purpose to hide `Flags_GetSwitch` calls behind a macro. Arguably, the code would be perfectly readable if `ENGO_CAGED_SWITCH_FLAG` was changed to `ENGO_IS_CAGE_OPEN_SWITCH_FLAG`